### PR TITLE
Upgrade to zio 2.0.1

### PR DESCRIPTION
**Pull request by vic**
Upgrade to latest ZIO 2.x release, during John's advanced-zio workshop :).

### CODE REVIEW

File: src/test/scala/zio/interop/TwitterSpec.scala [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
+object TwitterSpec extends ZIOSpecDefault {
   override val runner = {
     val ec       = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(2))
     val executor = Executor.fromExecutionContext(RuntimeConfig.defaultYieldOpCount)(ec)
 
     defaultTestRunner.withRuntimeConfig(_.copy(executor = executor))
   }
 
-  val runtime = runner.runtime
+  override val runtime: Runtime[Any] = Runtime.default // runner.runtime
```
Comment (vic) [inline src/test/scala/zio/interop/TwitterSpec.scala:22]: Revert change once we know how to build the runner in zio 2.

**Comment by vic** [comment]
Reminder for self: Also update README, since the `fromTwitterFuture` no longer lives on `Task` but on `ZIO` now. Also now it's `runtime.unsafe.runToTwitterFuture`.

**Comment by bpatters** [comment]
When do we think this will get merged? I needed this so had cut and paste the code from this MR to use it until this merged in. Anything I can do to help?

**Review by vigoo — APPROVED** [review]

**Comment by bpatters** [comment]
Any update on when this will be merged and released?

**Comment by bpatters** [comment]
@vigoo any update on when this will be merged and released?

**Pull request merged**